scheduler/cloudformation/cloudformation.go

```diff
@@ -280,13 +294,118 @@ func (s *Scheduler) submit(ctx context.Context, tx *sql.Tx, app *scheduler.App,
 	return nil
 }
 
-// updateStack performs a stack update, but waits for the stack to enter a
-// stable state before starting.
+func (s *Scheduler) createStack(input *cloudformation.CreateStackInput, done chan error) error {
+	p := func() error {
+		_, err := s.cloudformation.CreateStack(input)
+		return err
+	}
+	waiter := s.cloudformation.WaitUntilStackCreateComplete
+	locked := make(chan struct{})
+	return s.performStackOperation(*input.StackName, p, waiter, locked, done)
+}
+
+// updateStack enqueues the stack update, and waits a set amount of time for the
+// stack update to be submitted.
+func (s *Scheduler) updateStack(input *cloudformation.UpdateStackInput, done chan error) error {
+	locked := make(chan struct{})
+	errCh := make(chan error)
+
+	go func() {
+		p := func() error { return s.executeStackUpdate(input) }
+		waiter := s.cloudformation.WaitUntilStackUpdateComplete
+		errCh <- s.performStackOperation(*input.StackName, p, waiter, locked, done)
+	}()
+
+	var err error
+	select {
+	case <-time.After(lockWait):
+		// FIXME: At this point, we don't want to affect UX by waiting
```

Comment:
i think we should be publishing an event here, that way you get it via SNS if you have that setup, at the very least it should go to the app stream?